WordPress/gutenberg
# [RNMobile] Add new block type badge to block inserter list

* **Gutenberg:** Self
* **Gutenberg Mobile:** https://github.com/wordpress-mobile/gutenberg-mobile/pull/3720
* **WPiOS:** https://github.com/wordpress-mobile/WordPress-iOS/pull/16857
* **WPAndroid:** https://github.com/wordpress-mobile/WordPress-Android/pull/15038

## Description
Fixes #30627. Add new block type badge to the native block inserter list. A "new" badge is displayed for block types configured as new. The configuration resides in the JavaScript and is provided to the native host app for storage. Once a given block type is inserted, it is no longer considered new. 

https://github.com/WordPress/gutenberg/issues/33528 is planned to further improve this feature by having publishing multiple posts also remove the "new" badge from block types. That feature is not included in this PR. 

## How has this been tested?
1. Delete previous WP app installations from device. 
2. Install WPiOS/WPAndroid build from sibling PRs. 
3. Log into account and open the Gutenberg Editor. 
4. ℹ️ **Expected:** Block inserter tooltip displayed.<sup>1</sup> 
4. Tap + button to open block inserter. 
5. ℹ️ **Expected:** Observe a "new" badge on the Embed, Search, and Audio blocks. 
6. Inserting one of these blocks. 
7. Reopen the block inserter. 
8. ℹ️ **Expected:** The block inserted no longer has a "new" badge. 
9. Exit the Gutenberg Editor. 
10. Reopen the Gutenberg Editor. 
12. ℹ️ **Expected:** Block inserter tooltip not displayed. 
11. Open the block inserter. 
12. ℹ️ **Expected:** The block inserted still does not have a "new" badge, but the other unused block types do have a "new" badge. 

[1]: ⚠️ The Editor Onboarding features, including the "new" block badge, is only visible for user's within in the "cohort." This consists of 50% of users based upon the user's ID. If the user is not in the cohort, the "new" block badge should not show up. 

## Screenshots 
| iOS | Android |
| --- | --- |
| ![ios-new-block](https://user-images.githubusercontent.com/438664/128043293-446b3868-7356-4c02-98f3-a92611fd3685.PNG) | ![android-new-block](https://user-images.githubusercontent.com/438664/128042132-111011d9-1e6b-42fc-99aa-9d5439e7b006.jpg) |
| ![ios-new-block-dark](https://user-images.githubusercontent.com/438664/128043346-51be74ff-4dde-4492-b4d2-de96be518e68.PNG) | ![android-new-block-dark](https://user-images.githubusercontent.com/438664/128042089-fa336025-85d8-4fd6-80d4-02948bd08f90.jpg) |

## Types of changes
New feature

## Checklist:
